src/lisp/kernel/clos/streams.lisp

```diff
@@ -812,42 +816,33 @@ truename."))
 
 ;;; FILE-DESCRIPTOR
 
-(defmethod stream-file-descriptor :before (stream &optional (direction :input))
-  (multiple-value-bind (predicate kind)     
-      (case direction
-        (:input  (values 'input-stream-p  "input"))
-        (:output (values 'output-stream-p "output"))
-        (t
-         (error 'simple-type-error
-                :format-control "Not a valid direction, ~S; must be one of ~
-                                 :INPUT or :OUTPUT."
-                :format-arguments (list direction)
-                :datum direction
-                :expected-type '(member :input :output))))
-    (unless (funcall predicate stream)
-      (error 'simple-type-error
-             :format-control "Not an ~A stream, ~S, although ~S ~
-                              was provided as DIRECTION."
-             :format-arguments (list kind stream direction)
-             :datum stream
-             :expected-type `(satisfies ,predicate)))))
-
-(defmethod stream-file-descriptor (stream &optional direction)
-  (declare (ignore direction))
-  (bug-or-error stream 'stream-file-descriptor))
-
-(defmethod stream-file-descriptor ((stream two-way-stream) &optional (direction
-                                                                      :input))
-  (stream-file-descriptor
-   (case direction
-     (:input  (two-way-stream-input-stream stream))
-     (:output (two-way-stream-output-stream stream)))
-   direction))
-
-(defmethod stream-file-descriptor ((stream file-stream) &optional (direction
-                                                                   :input))
-  (declare (ignore direction))
-  (ext:file-stream-file-descriptor stream))
+(defmethod stream-file-descriptor (stream direction)
+  (declare (ignore stream direction))
+  nil)
+
+(defmethod stream-file-descriptor ((stream ansi-stream) direction)
+  (%stream-file-descriptor direction))
+
+
+;;; STREAM-INPUT-COLUMN
+
+(defmethod stream-input-column (stream)
+  (declare (ignore stream))
+  nil)
+
+(defmethod stream-input-column ((stream ansi-stream))
+  (%stream-input-column direction))
```

Comment:
Should have the stream rather than the direction, surely. If the tests don't cover this maybe they ought to.

Reply:
More copy/paste errors.